### PR TITLE
Feature/table filters

### PR DESCRIPTION
**Pull request by GuySerfaty**

### CODE REVIEW

**Review by amilich — COMMENTED** [review]
I think package-lock.json is unintentional?

File: src/columnsTypes/filtersLogic.js [hunk 1]
```diff
@@ -1,53 +1,60 @@
+import dayjs from 'dayjs';
+
+const clearWeirdCharactersFromText = (text) =>
```
Comment (amilich) [inline src/columnsTypes/filtersLogic.js:3]: name better? + document (lol)?

File: src/columnsTypes/filtersLogic.js [hunk 2]
```diff
@@ -97,97 +128,81 @@ const removeCurrencyFromText = (text) => {
...
+  if (!cell.firstChild.attrs.labels) return false;
+  const cellLabels = cell.firstChild.attrs.labels.map((label) => label.title);
+
+  for (let i = 0; i < value.length; i++) {
+    if (cellLabels.includes(value[i])) return true;
+  }
+  return false;
+};
+
+export const labelsIsNone = (cell, value) => {
```
Comment (amilich) [inline src/columnsTypes/filtersLogic.js:188]: bit of docs about this file/functions?

File: src/columnsTypes/types.config.js [hunk 3]
```diff
@@ -35,55 +73,302 @@ export const types = [
...
+      {
+        id: 'text-equality',
+        label: 'Is',
+        logic: textEquality,
+        defaultValue: '',
+        default: true,
+        inputType: 'input',
+      },
+      {
+        id: 'text-in-equality',
```
Comment (amilich) [inline src/columnsTypes/types.config.js:86]: "inequality"?

File: src/columnsTypes/types.config.js [hunk 4]
```diff
@@ -35,55 +73,302 @@ export const types = [
...
+      {
+        id: 'number-equality',
+        label: '=',
+        logic: numberEquality,
+        defaultValue: '',
+        default: true,
+        inputType: 'input',
+      },
+      {
+        id: 'number-in-equality',
```
Comment (amilich) [inline src/columnsTypes/types.config.js:134]: "inequality"?

File: src/columnsTypes/types.config.js [hunk 5]
```diff
@@ -35,55 +73,302 @@ export const types = [
...
+      {
+        id: 'currency-equality',
+        label: '=',
+        logic: currencyEquality,
+        defaultValue: '',
+        default: true,
+        inputType: 'input',
+      },
+      {
+        id: 'currency-in-equality',
```
Comment (amilich) [inline src/columnsTypes/types.config.js:269]: same here

File: src/columnsTypes/types/Label/Component.jsx [hunk 6]
```diff
@@ -78,6 +97,8 @@ export const LabelsChooser = ({view, pos, node}) => {
   );
 
   const handleClose = (currentChosenLabels) => {
+    if (onClose) onClose();
```
Comment (amilich) [inline src/columnsTypes/types/Label/Component.jsx:100]: is the below a default on close procedure?

File: src/commands.js [hunk 7]
```diff
@@ -278,13 +278,16 @@ export function deleteRow(state, dispatch) {
       tr = state.tr;
     if (rect.top == 0 && rect.bottom == rect.map.height) return false;
     for (let i = rect.bottom - 1; ; i--) {
+      const row = rect.table.child(i)
+      if(row.attrs.hidden) continue;
```
Comment (amilich) [inline src/commands.js:282]: nit - space `if (`

File: src/filters/Component.jsx [hunk 8]
```diff
@@ -0,0 +1,377 @@
...
+  switch (inputType) {
+    default:
+      return () => <></>;
+    case 'input':
+      return FiltersTextInput;
+    case 'date-picker':
+      return FiltersDatePicker;
+    case 'dropdown':
+      return FiltersInputDropDown;
+    case 'labels-dropdown':
```
Comment (amilich) [inline src/filters/Component.jsx:146]: do we want an enum for this? maybe could be shared with `types.config`?

File: src/tooltip-menus/items.js [hunk 9]
```diff
@@ -6,12 +6,9 @@ import {
...
   enableCellsColor,
 } from './utils';
-import {
-  getDeleteCommand,
-  changeCellsBackgroundColor,
-  toggleTableHeaders,
-} from './commands';
+import {changeCellsBackgroundColor, toggleTableHeaders} from './commands';
 import {createElementWithClass} from '../util';
+import {getDeleteCommand} from '../input';
```
Comment (amilich) [inline src/tooltip-menus/items.js:11]: nit - spaces
might want to lint at this point, this library is becoming very complex :)

File: src/menuItems/MenuItem.js [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+  }
+  const sym = document.createElementNS(SVG, 'symbol');
+  sym.id = name;
+  sym.setAttribute('viewBox', '0 0 ' + data.width + ' ' + data.height);
+  const path = sym.appendChild(document.createElementNS(SVG, 'path'));
+  path.setAttribute('d', data.path);
+  collection.appendChild(sym);
+}
+
+export function getIcon(icon) {
```
Comment (amilich) [inline src/menuItems/MenuItem.js:41]: document how this works? do we want to use our own svgs?

File: src/input.js [hunk 11]
```diff
@@ -58,35 +64,186 @@ function maybeSetSelection(state, dispatch, selection) {
...
+    if (index != (dir < 0 ? 0 : parent.childCount)) return null;
+    if (
+      parent.type.spec.tableRole == 'cell' ||
+      parent.type.spec.tableRole == 'header_cell'
+    ) {
+      const cellPos = $head.before(d);
+      const dirStr =
+        axis == 'vert' ? (dir > 0 ? 'down' : 'up') : dir > 0 ? 'right' : 'left';
+
+      return (axis == 'vert'
```
Comment (amilich) [inline src/input.js:193]: document?

File: src/input.js [hunk 12]
```diff
@@ -58,35 +64,186 @@ function maybeSetSelection(state, dispatch, selection) {
...
+
+// : (EditorView, number, number)
+// Whether vertical position motion in a given direction
+// from a position would leave a text block.
+function endOfTextblockVertical(view, state, dir, selection) {
+  const sel = selection;
+  const $pos = dir === 'up' ? sel.$from : sel.$to;
+  return withFlushedState(view, state, () => {
+    let {node: dom} = view.docView.domFromPos($pos.pos, dir == 'up' ? -1 : 1);
+    for (;;) {
```
Comment (amilich) [inline src/input.js:100]: how does this work?

File: src/filters/DropDown.jsx [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+  const [showDropDown, setShowDropDown] = useState(false);
+  const dropDownRef = useRef();
+
+  const closeDropDown = useCallback(() => {
+    setShowDropDown(false);
+    if (
+      dropDownRef.current &&
+      dropDownRef.current.classList.contains('open-up')
+    ) {
+      dropDownRef.current.classList.remove('open-up');
```
Comment (amilich) [inline src/filters/DropDown.jsx:88]: nit - rename as `Dropdown` (file + vars?)

File: src/filters/DropDown.jsx [hunk 14]
```diff
@@ -0,0 +1,156 @@
+import React, {useState, useRef, useCallback, useMemo} from 'react';
+
+// export type DropDownItemType = {
```
Comment (amilich) [inline src/filters/DropDown.jsx:3]: no types?

**Review by amilich — COMMENTED** [review]
A few really small comments. The code looks very clean.

File: src/columnsTypes/types/Label/Component.jsx (hunk 6 above)
Comment (OfirSimhi1612) [reply src/columnsTypes/types/Label/Component.jsx:100]: yep, added a comment.
the labels dropdown can "live" in two places - labels cell and filters, each one has different close procedure

File: src/filters/Component.jsx (hunk 8 above)
Comment (OfirSimhi1612) [reply src/filters/Component.jsx:146]: were still working with js here so enum is not an option for now. might be nice improvement to make it a ts repo :)  
we can make some kind of map if it makes more sense and feels more natural

File: src/tooltip-menus/items.js (hunk 9 above)
Comment (OfirSimhi1612) [reply src/tooltip-menus/items.js:11]: sounds good, adding eslint to the bugs cycle branch is ok? or we want it in this pr already?

File: src/menuItems/MenuItem.js (hunk 10 above)
Comment (OfirSimhi1612) [reply src/menuItems/MenuItem.js:41]: we added some copied code from prosemirror core libraries so we can make minor changes the library is not supporting.
this function basically build an icon element for menu items according to a spec passed to her. were not really using it, just needed it as dependency for another function

File: src/input.js (hunk 11 above)
Comment (OfirSimhi1612) [reply src/input.js:193]: same as `getIcon`, will add documentation to the parts we changed

File: src/input.js (hunk 12 above)
Comment (OfirSimhi1612) [reply src/input.js:100]: same here

File: src/filters/DropDown.jsx (hunk 14 above)
Comment (OfirSimhi1612) [reply src/filters/DropDown.jsx:3]: same as the enum, saving it for a day we will write ts here :)

File: src/filters/Component.jsx (hunk 8 above)
Comment (amilich) [reply src/filters/Component.jsx:146]: sounds good! can add a ticket if you see necessary

File: src/tooltip-menus/items.js (hunk 9 above)
Comment (amilich) [reply src/tooltip-menus/items.js:11]: yes - can be a new ticket!

File: src/menuItems/MenuItem.js (hunk 10 above)
Comment (amilich) [reply src/menuItems/MenuItem.js:41]: got it

File: src/input.js (hunk 12 above)
Comment (amilich) [reply src/input.js:100]: just wondering how this for loop `for (;;)` works?

Comment (amilich) [reply src/input.js:100]: is this an infinite loop? @ehsanmasdar

Comment (OfirSimhi1612) [reply src/input.js:100]: yes 👍🏽

**Pull request merged**